sdks/java/testing/test-utils/build.gradle

```diff
@@ -40,15 +40,15 @@ dependencies {
   provided library.java.jmh_core
 
   testImplementation library.java.junit
-  testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadowTest")
+  testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadow")
 }
 
-['8', '11', '17', '21'].each {
-  tasks.create(name: "verifyJavaVersion${it}", type: Test) {
+['8', '11', '17', '21'].each { String ver ->
+  tasks.create(name: "verifyJavaVersion${ver}", type: Test) {
```

Comment:
Groovy has a many implicit and errorprone syntax and this is one of it. Foreach closure has an implicit variable that has default name `it`. However, the tasks.create closure override the same name, causing "${it}" resolved to "Task " and as a result, only verifyCodeIsCompiledWithJava8 was run

Reply:
oof. good catch.